packages/editor/src/store/reducer.js

```diff
@@ -887,9 +886,11 @@ export function blockSelection( state = {
 				return state;
 			}
 
-			// If there is replacement block(s), assign first's client ID as
-			// the next selected block. If empty replacement, reset to null.
-			const nextSelectedBlockClientId = get( action.blocks, [ 0, 'clientId' ], null );
+			// If there are replacement blocks, assign last block as the next
+			// selected block, otherwise set to null.
+			const lastBlock = last( action.blocks );
+			const nextSelectedBlockClientId = lastBlock ? lastBlock.clientId : null;
+
```

Comment:
if we put `initialPotition` to `-1` when selecting the last block, I guess this would put the caret at the end two?

Reply:
Tests will fail. See https://github.com/WordPress/gutenberg/pull/13294#issue-244078254 (issue description).

Reply:
Got it, I guess pasting needs to be a special treatment in that case.